### ISSUE TITLE
Perplexity unequivalent when training example = validation example

### ISSUE THREAD

**Issue by janenie** [OpenNMT/OpenNMT-py#625]
Hi,
I found a weird thing, I use only one training example, and set validation set to the only training example
I tune a few parameters (I use GRU for both enc/dec, and adadelta for optimization, max_grad_norm is 5)

I found that after >=200 epoches, the training perplexity is only 1.7 (it should be 1 actually), and validation perplexity is 1.4 (it should be equivalent to training perplexity, since validation set is equivalent to training set). I also set the copy_force=False, since it may add an additional copy loss (If I understand the code correctly)

Why is this happening?

**Comment by srush** [comment]
There are certain features such as `dropout` that will be on during training and off during validation. Be sure your model does not have those on.

**Comment by janenie** [comment]
Yes, turn off the dropout will make the ppl is equivalent on dev/train, but I notice that whenever I add dropout option, the ppl is hardly descrease to 1, while there are only 1 training/testing cases. When I set the dropout to 0, with special hyper parameter, it might converges to 1. I wonder if there is something I've been missing when adding dropout
